subprojects/docs/src/docs/userguide/working_with_files.adoc

```diff
@@ -364,173 +396,190 @@ File collections have some important attributes in Gradle. They can be:
 
-++++
-<figure>
-    <title>The differences in how file trees and file collections behave when copying files</title>
-    <imageobject>
-        <imagedata fileref="img/file-collection-vs-file-tree.png" width="160mm"/>
-    </imageobject>
-</figure>
-++++
+.The differences in how file trees and file collections behave when copying files
+image::file-collection-vs-file-tree.png[]
 
-NOTE: Although `FileTree` extends `FileCollection` (an is-a relationship), their behaviors do differ. In other words, you can use a file tree wherever a file collection is required, but remember: a file collection is a flat list/set of files, while a file tree is a file and directory hierarchy. To convert a file tree to a flat collection, use the api:org.gradle.api.file.FileTree#getFiles()[] property.
+NOTE: Although `FileTree` extends `FileCollection` (an is-a relationship), their behaviors do differ. In other words, you can use a file tree wherever a file collection is required, but remember: a file collection is a flat list/set of files, while a file tree is a file and directory hierarchy. To convert a file tree to a flat collection, use the link:{javadocPath}/org/gradle/api/file/FileTree.html#getFiles--[FileTree.getFiles()] property.
 
-The simplest way to create a file tree is to pass a file or directory path to the api:org.gradle.api.Project#fileTree(java.lang.Object)[] method. This will create a tree of all the files and directories in that base directory (but not the base directory itself). The following example demonstrates how to use the basic method and, in addition, how to filter the files and directories using Ant-style patterns:
+The simplest way to create a file tree is to pass a file or directory path to the link:{groovyDslPath}/org.gradle.api.Project.html#org.gradle.api.Project:fileTree(java.lang.Object)[Project.fileTree(java.lang.Object)] method. This will create a tree of all the files and directories in that base directory (but not the base directory itself). The following example demonstrates how to use the basic method and, in addition, how to filter the files and directories using Ant-style patterns:
 
-++++
-<sample id="fileTrees" dir="userguide/files/fileTrees" title="Creating a file tree">
-    <sourcefile file="build.gradle" snippet="define"/>
-</sample>
-++++
+=== Example: Creating a file tree
 
-You can see more examples of supported patterns in the API docs for api:org.gradle.api.tasks.util.PatternFilterable[]. Also, see the API documentation for `fileTree()` to see what types you can pass as the base directory.
+[source.multi-language-sample,groovy]
+.build.gradle
+----
+include::{samplesPath}/userguide/files/fileTrees/build.gradle[tag=define]
+----
+
+You can see more examples of supported patterns in the API docs for link:{javadocPath}/org/gradle/api/tasks/util/PatternFilterable.html[PatternFilterable]. Also, see the API documentation for `fileTree()` to see what types you can pass as the base directory.
 
-[NOTE]
```

Comment:
This `NOTE` element is totally missing in the converted version.

Reply:
This is intentional.